plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+	IndexesAllocatedDataFileInMb   int64 `json:"IndexesAllocatedDataFileInMb"`
+	IndexesUsedDataFileInMb        int64 `json:"IndexesUsedDataFileInMb"`
+	TotalAllocatedStorageFileInMb  int64 `json:"TotalAllocatedStorageFileInMb"`
+	TotalFreeSpaceInMb             int64 `json:"TotalFreeSpaceInMb"`
+}
+
+type indexesMetricResponse struct {
+	Results         []*indexMetrics `json:"Results"`
+	PublicServerUrl string          `json:"PublicServerUrl"`
+	NodeTag         string          `json:"NodeTag"`
+}
+
+type indexMetrics struct {
+	DatabaseName               string   `json:"DatabaseName"`
+	IndexName                  string   `json:"IndexName"`
+	Priority                   string   `json:"Priority"`
+	State                      string   `json:"State"`
+	Errors                     int32    `json:"Errors"`
+	TimeSinceLastQueryInSec    *float64 `json:"TimeSinceLastQueryInSec"`
+	TimeSinceLastIndexingInSec *float64 `json:"TimeSinceLastIndexingInSec"`
+	LockMode                   string   `json:"LockMode"`
+	IsInvalid                  bool     `json:"IsInvalid"`
+	Status                     string   `json:"Status"`
+	MappedPerSec               float64  `json:"MappedPerSec"`
+	ReducedPerSec              float64  `json:"ReducedPerSec"`
+	Type                       string   `json:"Type"`
+	EntriesCount               int32    `json:"EntriesCount"`
+}
+
+type collectionsMetricResponse struct {
+	Results         []*collectionMetrics `json:"Results"`
+	PublicServerUrl string               `json:"PublicServerUrl"`
+	NodeTag         string               `json:"NodeTag"`
+}
+
+type collectionMetrics struct {
+	DatabaseName          string `json:"DatabaseName"`
+	CollectionName        string `json:"CollectionName"`
+	DocumentsCount        int64  `json:"DocumentsCount"`
+	TotalSizeInBytes      int64  `json:"TotalSizeInBytes"`
+	DocumentsSizeInBytes  int64  `json:"DocumentsSizeInBytes"`
+	TombstonesSizeInBytes int64  `json:"TombstonesSizeInBytes"`
+	RevisionsSizeInBytes  int64  `json:"RevisionsSizeInBytes"`
+}
+
+var sampleConfig = `
+  ## Node URL and port that RavenDB is listening on.
+  url = "https://localhost:8080"
+
+  ## RavenDB X509 client certificate setup
+  tls_cert = "/etc/telegraf/raven.crt"
+  tls_key = "/etc/telegraf/raven.key"
+
+  ## Optional request timeouts
+  ##
+  ## ResponseHeaderTimeout, if non-zero, specifies the amount of time to wait
+  ## for a server's response headers after fully writing the request.
+  # header_timeout = "3s"
+  ##
+  ## client_timeout specifies a time limit for requests made by this client.
+  ## Includes connection time, any redirects, and reading the response body.
+  # client_timeout = "4s"
+
+  ## When true, collect server stats
+  # gather_server_stats = true
+
+  ## When true, collect per database stats
+  # gather_db_stats = true
+
+  ## When true, collect per index stats
+  # gather_index_stats = true
+
+  ## When true, collect per collection stats
+  # gather_collection_stats = true
+
```

Comment:
Does it make sense to have a selection list instead like `stats_include=["server", "db", "index",...]`? This is much easier to extend... You might event add an exclude list and gather everything by default if `stats_include` is empty. Maybe take a look at `plugins/inputs/cloudwatch` as a good example.

Reply:
since we have 4 switches I think let's leave includes only - I removed boolean - I agree this is redundant.

Reply:
ok I just recall why we have bool + list. If we don't want to collect given stats then we set boolean to false. Having empty list might be ambiguous (does it mean collect all items are none?). Please have a look at mongodb input plugin. I follow the same pattern.

Reply:
Well, lately we had a lot of troubles with other plugins that started to extend the options or make the selection more fine-grained. You can easily end-up with dozens of options in the worst-case. A list is easy to extend and you can define and document the empty behavior. Most plugins (like `cloudwatch`) define an `_include` and an `_exclude` list with empty include-list means _all_ included.

So if there is a chance (even a small one) that at some point you want to add/split/remove items here, please go with the list. There is even a built-in `filter` implementation that makes using such lists dead easy (including glob patterns).

Reply:
and what about disabling given feature? Should user do: collections_include = ["THIS_COLLECTION_DOES_NOT_EXIST"] ?

Maybe (it this is common solution), we could do:
nil -> get nothing
empty -> get all
some items -> get only those items? 

But such mapping is always problematic. That the reason I like that in GO string cannot be null (unless we use pointers)

Reply:
No please don't invent anything new. Look at your `*StatsDbs` lists, there you also define a behavior for an empty list. To make an example, currently you have the possible values `server`, `index`, `collection` for the `stats_include` list. To match your current default this corresponds to a
```
stats_include = ["server", "index", "collection"]
```
setting in the config. To disable a certain feature, just remove it from the list. To e.g. disable `collection` just specify 
```
stats_include = ["server", "index"]
```

Now you only have to care for the case of the user specifying an empty list `stats_include = []`. There are two options.
1. An empty list means **nothing** is collected. You might output a warning or error out in this case. If you decide for this behavior, you can set the default `stats_include = ["server", "index", "collection"]` when initializing the structure in `init()`.
2. You can define an empty list as **everything** is collected as a number of other plugins do. This is beneficial if you have many possible options and the default case is to collect everything. In this case you might check the value of `StatsInclude` in `Init()` and fill in all options there to not clutter the `Gather()` function.

As you see the most difficult part is to decide which behavior you want for an empty list. Using `internal.choice` makes it easy to check if the options are valid or if a certain option is contained. In case you define _empty means all_, you can even accompany the `stats_include` with an `stats_exclude`, but for me this is overdoing for three options.